gammapy/spectrum/core.py

```diff
@@ -77,19 +77,11 @@ def from_hdulist(cls, hdulist, hdu1='COUNTS', hdu2='EBOUNDS'):
                    energy_hi=ebounds.upper_bounds)
 
     @classmethod
-    def read(cls, filename, hdu1='COUNTS', hdu2='EBOUNDS', **kwargs):
+    def read(cls, filename, hdu1='COUNTS', hdu2='EBOUNDS'):
         """Read from file."""
         filename = make_path(filename)
-        with fits.open(str(filename), **kwargs) as hdulist:
-            try:
-                spec = cls.from_hdulist(hdulist, hdu1=hdu1, hdu2=hdu2)
-            except KeyError:
-                msg = 'File {} does not contain HDUs "{}"'.format(
-                    filename, [hdu1, hdu2])
-                msg += '\n Available {}'.format([_.name for _ in hdulist])
-                raise ValueError(msg)
-
-        return spec
+        with fits.open(str(filename), memmap=False) as hdulist:
+            return cls.from_hdulist(hdulist, hdu1=hdu1, hdu2=hdu2)
```

Comment:
@joleroi - Was there a good reason to catch the KeyError and raise a ValueError instead?

My suggestion would be that here in gammapy.spectrum we just let the KeyError appear if users try to access HDUs that aren't there; otherwise we get mixed behaviour / errors from Gammapy read methods.